ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js

```diff
@@ -0,0 +1,512 @@
+          .send({ from: address, value: value, })
+        return res.status(200).json({ result })
+      } catch (e) {
+        console.error(e);
+        return res.status(500).json({ errors: [e] });
+      }
+    }
+  );
+
+
+
+  router.post(
+    "/:address/swap/token", [
+    param("address").trim().isString(),
+    query("tokenFrom").trim().isIn(["dai", "mkr", "uni"]),
+    query("tokenTo").trim().isIn(["eth"]),
+    query("privateKey").trim().isString(),
+  ],
+    async (req, res, next) => {
+      try {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+          return res.status(400).json({ errors: errors.array() });
+        }
+        const { address = "*" } = req.params;
+        const { privateKey = "", amount = "", tokenFrom = "", tokenTo = "" } = { ...req.query };
+        const tokenType = tokenFrom.toString().toUpperCase();
+        const routerAbi = abi.UNISWAP_ROUTER.abi
+        const tokenAbi = abi.APIS_ERC20
+        const chainIds = ChainId.RINKEBY;
+
+        const tokenAddressRinkebyDai = '0xc7ad46e0b8a400bb3c915120d284aafba8fc4735'
+        const tokenAddressRinkebyUniswap = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984';
+        const tokenAddressRinkebyMaker = '0xF9bA5210F91D0474bd1e1DcDAeC4C58E359AaD85';
+        const tokenAddressRinkebyRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+
+        const DAI = await Fetcher.fetchTokenData(chainIds, tokenAddressRinkebyDai);
+        const UNISWAP = await Fetcher.fetchTokenData(chainIds, tokenAddressRinkebyUniswap);
+        const MAKER = await Fetcher.fetchTokenData(chainIds, tokenAddressRinkebyMaker);
+        const weth = WETH[chainIds]
+
+        let token = {};
+        switch (tokenType) {
+          case 'DAI':
+            token = DAI;
+            break;
+          case 'UNI':
+            token = UNISWAP;
+            break;
+          case 'MKR':
+            token = MAKER
+            break;
+        }
+        if (!Object.keys(token).length) {
+          return res.status(500).json({ error: "token value cannot be empty" })
+        }
+        const pair = await Fetcher.fetchPairData(token, weth)
+        const route = new Route([pair], token)
+        const amountIn = web3.utils.toWei(amount);
+        const trade = new Trade(route, new TokenAmount(token, amountIn), TradeType.EXACT_INPUT)
+        const slippageTolerance = new Percent('50', '10000')
+        const amountOutMin = trade.minimumAmountOut(slippageTolerance).raw
+        const path = [token.address, weth.address]
+        const deadline = Math.floor(Date.now() / 1000) + 60 * 20
+        const valueInput = trade.inputAmount.raw
+        const amountOutMinHex = ethers.BigNumber.from(amountOutMin.toString()).toHexString();
+        const value = ethers.BigNumber.from(valueInput.toString()).toHexString();
+        const wallet = web3.eth.accounts.wallet.add({
+          privateKey,
+          address,
+        });
```

Comment:
Many of these should be set as default values for user-provided parameters.

Reply:
will consider in the future implementations